server/core/runtime/models/shell_command_runner.go

```diff
@@ -147,13 +149,16 @@ func (s *ShellCommandRunner) RunCommandAsync(ctx command.ProjectContext) (chan<-
 		// Wait for the command to complete.
 		err = s.cmd.Wait()
 
+		dur := time.Since(start)
```

Comment:
Please add a unit test for this in `shell_command_runner_test.go`. if this doesn't exist, please create one to ensure this functionality works as expected and doesn't break in the future.